Implement user metadata API/CLI

mweibel opened the pull request.

dpacassi commented on rokka/sourceimages.go:
```diff
@@ -232,3 +232,64 @@ func (c *Client) DeleteDynamicMetadata(org, hash, name string, options DynamicMe
+	}
+
+	return c.Call(req, nil, nil)
+}
```
The SetUserMetadata, UpdateUserMetadata and DeleteUserMetadata functions look quite similar.
Maybe we should create a helper function which gets called by these 3 functions with the right parameters in order to reduce redundancy?

mweibel replied:
fixed

dpacassi commented on rokka/sourceimages.go:
```diff
@@ -232,3 +232,64 @@ func (c *Client) DeleteDynamicMetadata(org, hash, name string, options DynamicMe
+	}
+
+	return c.Call(req, nil, nil)
+}
```
Same goes for the UpdateUserMetadataByName and DeleteUserMetadataByName functions. Maybe we should create a helper function which gets called by these 2 functions with the right parameters in order to reduce redundancy?

mweibel replied:
fixed, too

mweibel commented:
@dpacassi I fixed the review comments and added another commit which fixes an issue with the create organization call (plus adds a test for it ;)) --> please don't squash if you merge.
